calico/felix/felix.py

```diff
@@ -159,29 +160,36 @@ def _main_greenlet(config):
 
 
 def main():
-    try:
-        # Initialise the logging with default parameters.
-        common.default_logging()
+    # Initialise the logging with default parameters.
+    common.default_logging()
 
-        # Create configuration, reading defaults from file if it exists.
+    # Create configuration, reading defaults from file if it exists.
+    try:
+        parser = optparse.OptionParser()
+        parser.add_option('-c', '--config-file', dest='config_file',
+                          help="configuration file to use",
+                          default="/etc/calico/felix.cfg")
+        options, args = parser.parse_args()
+        config = Config(options.config_file)
```

Comment:
I still think this `try` block is overbroad. Errors from optparse should not be caught, regardless of what they are, so let's restrict the `try` block to only the `config = Config(options.config_file)` line.